### ISSUE TITLE
Formatter ignores 'numberFormatterOptions' when asCurrency() is used.

### ISSUE THREAD

**Issue by ptz-nerf** [yiisoft/yii2#14874]
### What steps will reproduce the problem?
```php
$formatter = new \yii\i18n\Formatter([
    'locale' => 'en-US',
    'numberFormatterOptions' => [
        \NumberFormatter::MIN_FRACTION_DIGITS => 0,
        \NumberFormatter::MAX_FRACTION_DIGITS => 2,
    ],
    'thousandSeparator' => ' ', // space
]);

echo $formatter->asCurrency(1000.5, 'USD');
echo $formatter->asDecimal(1000.5);
```
### What is the expected result?
$1 000.5
1 000.5
### What do you get instead?
$1 000.50 <== !!!
1 000.5

numberFormatterOptions are ignored while using asCurrency().
see https://github.com/yiisoft/yii2/issues/12345#issuecomment-331181083

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 5.6
| Intl version      | 1.1.0, PECL-3.0.0
| Operating system | Windows 10, CentOS

**Comment by SilverFire** [comment]
What is your ICU Data version?

**Comment by ptz-nerf** [comment]
CentOS - 58.2 (PECL-3.0.0)
Windows - 57.1 (1.1.0)

**Comment by cebe** [comment]
this is because fraction digits are overwritten by the currency. If you set currency on the formatter it will set fraction digits dependent on the currency. What is the use case to format currency in a non-standard way? people usually would expect dollars to be rendered as $1,50  and not $1,5.

**Comment by ptz-nerf** [comment]
People usually would expect dollars to be rendered as $150 and not $150.00!

**Comment by rob006** [comment]
@ptz-nerf
> Correct, but people usually would expect dollars to be rendered as $150 and not $150.00!

Not really. I work a lot with indexing prices in online stores and all of them displays all fraction digits even if there are only zeros. The only exception are currencies with a high denomination like RUB - for them you never have any fraction digits. 
Returning something like `$1 000.5` will be definitely a bug for me.

**Comment by ptz-nerf** [comment]
@rob006 I agree. But we have inconsistency here. There are two normal options:
- asCurrency() ignores Formatter settings and use only locale settings,
- does not ignore setting that are explicitly specified.

**Comment by rob006** [comment]
I don't think that `asCurrency()` should use `numberFormatterOptions` from formatter. Actually there are only two valid settings for currencies formatting:
* Always add fraction digits (`$9.99` and `$10.00`).
* Add fraction digits only for "non-integer" numbers  (`$9.99` and `$10`).

Configuring this by `MIN_FRACTION_DIGITS` and `MAX_FRACTION_DIGITS` doesn't make sense.

**Comment by PendalF89** [comment]
Hi! I have a same problem in 2.0.16, ICU version is 60.2, PHP version is 7.2.15, OS Ubuntu 18,04.
`echo Yii::$app->formatter->asCurrency(2000, 'USD'); // "2 000,00 $", but except "2 000 $"`

**Comment by PendalF89** [comment]
I know, that's is bad decision, but i solved my problem as follows:

1. Inherited the `yii\i18n\Formatter` class
2. Rewrite `asCurrency` method:

```
public function asCurrency($value, $currency = null, $options = [], $textOptions = [], $stripUnnecessaryFractionDigits = true)
{
	$result = parent::asCurrency($value, $currency = null, $options = [], $textOptions = []);
	if ($stripUnnecessaryFractionDigits) {
		$result = str_replace(',00', '', $result);
	}
	return $result;
}
```

**Comment by terabytesoftw** [comment]
I could not reproduce this on current Yii2 master (May 2026).

The original reproduction now returns `$1 000.5` for `asCurrency(1000.5, 'USD')` and `1 000.5` for `asDecimal(1000.5)`, so `numberFormatterOptions` are applied to currency formatting. The related currency fraction tests (`testIntlCurrencyFraction` and `testIntlIssue5261`) also pass.

The current implementation sets the currency code before applying `NumberFormatter` attributes and merges `numberFormatterOptions` in `createNumberFormatter()`, so the reported behavior appears to be fixed already.

Closing as not reproducible in the current version.
